pandas/core/series.py

```diff
@@ -1110,6 +1113,10 @@ def _set_value(self, label, value, takeable: bool = False):
             else:
                 loc = self.index.get_loc(label)
                 validate_numeric_casting(self.dtype, value)
+                dtype, _ = infer_dtype_from_scalar(value, pandas_dtype=True)
+                if not is_dtype_equal(self.dtype, dtype) and is_numeric_dtype(dtype):
```

Comment:
why do you need to check if is numeric here? isn't just not dtype equal enough?

Reply:
We have a test calling this function with scalar is a string value, which is expected to raise an ValueError, hence this check there. If the test is somehow erroneous, we could delete it.

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=47080&view=logs&j=ba13898e-1dfb-5ace-9966-8b7af3677790
The failing test is in this build

Reply:
can you show the test

Reply:
``test_set_value_resize`` in ``<pandas.tests.frame.indexing.test_set_value``.

Raising lines are

https://github.com/pandas-dev/pandas/blob/c5d105b1a32a02cce0e68ff964b6b30e9d682506/pandas/tests/frame/indexing/test_set_value.py#L39:L40